pkg/authn/identityheaders/identityheaders.go

```diff
@@ -52,7 +50,11 @@ func WithAuthHeaders(handler http.Handler, cfg *AuthnHeaderConfig) http.Handler
 			// Seemingly well-known headers to tell the upstream about user's identity
 			// so that the upstream can achieve the original goal of delegating RBAC authn/authz to kube-rbac-proxy
 			req.Header.Set(cfg.UserFieldName, u.GetName())
-			req.Header.Set(cfg.GroupsFieldName, strings.Join(u.GetGroups(), cfg.GroupSeparator))
```

Comment:
Does this drop the `GroupSeparator`?

While I am personally in favour of using the HTTP-standard header separation, I believe the original review comment requested a bit of a different outcome: https://github.com/brancz/kube-rbac-proxy/pull/229/files#diff-1ca0b12c9f8ec6aec4fea4c5542fe1bb7156876c19d791d1fcf7cf66e62905aeR57-R61. This is to support services that expect different groups formatting.

I suggest:
1. by default, do the right thing and behave like a good HTTP client (=> what you have here)
2. if a group separator is set, do what the code does currently.

On top of the above, we also need to:
- figure out what to do with groups that contain the standard separator ("," according to https://www.rfc-editor.org/rfc/rfc9110.html#name-header-fields) in the standard case - what does the Golang code do in that case?
- drop the groups containing the group separator as the original comment suggests in the `groupSeparator != ""` case

Reply:
Thank you for making that clear! I've updated the code to address it and updated the unit tests to cover these scenarios